tests/python/tank_test/tank_test_base.py

```diff
@@ -392,10 +397,17 @@ def _mock_return_value(self, to_mock, return_value):
         patch.start()
         self.addCleanup(patch.stop)
 
+    def _assert_teardown_called(self):
```

Comment:
If a test doesn't tear down it can have disastrous consequences for the tests that follows it. A test should fail if it fails at tearing down the test.